datumaro/cli/__main__.py

```diff
@@ -131,18 +153,29 @@ def main(args=None):
         parser.print_help()
         return 1
 
+    args_with_paths = _get_sensitive_args()
+    telemetry = init_telemetry_session(app_name='Datumaro', app_version=VERSION)
+
     try:
         retcode = args.command(args)
         if retcode is None:
             retcode = 0
-        return retcode
     except CliException as e:
         log.error(e)
+        send_command_exception_info(telemetry, args, args_with_paths[args.command])
         return 1
     except Exception as e:
         log.error(e)
+        send_command_exception_info(telemetry, args, args_with_paths[args.command])
         raise
-
+    else:
+        if retcode:
+            send_command_failure_info(telemetry, args, args_with_paths[args.command])
```

Comment:
Both `except` branches won't send failure notification. Probably, it should be excluded from `else` and moved to `finally`?